### PR TITLE
jetbrains.gateway: init at 213.5744.213

### PR DESCRIPTION
**Pull request by dev-null-undefined**
###### Motivation for this change

Addition of JetBrains gateway IDE

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by VergeDX — APPROVED** [review]

**Comment by anoadragon453** [comment]
Another PR that adds this same package: https://github.com/NixOS/nixpkgs/pull/152009 (albeit at a different version).

File: pkgs/applications/editors/jetbrains/default.nix [hunk 1]
```diff
@@ -69,6 +69,24 @@ let
...
+      meta = with lib; {
+        homepage = "https://www.jetbrains.com/remote-development/gateway/";
+        inherit description license;
+        longDescription = ''
+          Experience a new compact application that connects you to
+          any of your remote machines  via SSH without any hassle.
+          It serves as a single entry point to all your remote work,
+          and you don’t even need to have a JetBrains IDE installed.
+        '';
+        maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/jetbrains/default.nix:85]: You need to add yourself here

File: pkgs/applications/editors/jetbrains/default.nix [hunk 2]
```diff
@@ -69,6 +69,24 @@ let
       };
     });
 
+  buildGateway = { name, version, src, license, description, wmClass, ... }:
+    (mkJetBrainsProduct {
+      inherit name version src wmClass jdk;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/jetbrains/default.nix:74]: is name latter converted to pname? If not that should be fixed in another PR

**Comment by anoadragon453** [comment]
I have made some progress towards a more up-to-date changeset in https://github.com/anoadragon453/nixpkgs/commits/anoa/jetbrains-gateway-222.3153.1, which people can make use of.

The second most recent commit (e42f77fc519875c4605bd106ed859d1835e54df4) inits JetBrains Gateway at 222.3153.1. However, I found that this version required a version of [JetBrains Runtime](https://github.com/JetBrains/JetBrainsRuntime/releases) of at least v17.*. Nixpkgs looks to [only have v11.*](https://github.com/NixOS/nixpkgs/blob/ccf8bdf72624521358be6bb7d9b524c4cbcf7aff/pkgs/development/compilers/jetbrains-jdk/default.nix#L1) at the moment.

So the most recent commit (4c6c136e01a14d85f16f505cb4e0f5c5df383357) instead inits JetBrains Gateway 213.6777.25. This compiles and installs successfully, though I couldn't get it working with a running instance of gitpod. I was unable to determine if this was an issue with my credentials or the software installation though.

**Comment by Mic92** [comment]
done in https://github.com/NixOS/nixpkgs/pull/195132
